### PR TITLE
Add the ability to mock the RECONNECT message

### PR DESCRIPTION
**Pull request by pajlada**
https://dev.twitch.tv/docs/irc/commands#reconnect-twitch-commands

This may be more targeted towards Twitch IRC library creators than bot creators, but it's a useful response to test nonetheless

### CODE REVIEW

**Comment by trezy** [comment]
@pajlada Sorry I didn't see this sooner! I'll pull it down and give it a peek this afternoon, then hopefully get it merged! 😁

**Comment by trezy** [comment]
@pajlada The only issue I have right now is that the `RECONNECT` message is currently handled the same way normal messages are, rather than the way that system messages are. For example, if you send a `/ping` command, it's handled specially by the [`handlePINGMessage`](https://github.com/fdgt-apis/api/blob/main/src/helpers/handlePINGMessage.js) helper. With the way that the reconnect is being handled in your PR, the response gets pushed through the `renderMessage` helper unnecessarily, causing a superfluous `@` to be appended to the response.

I'd prefer that we move the handling of the reconnects into a method on the connection (`connection.sendReconnect`, similar to [`connection. sendUnknownCommand`](https://github.com/fdgt-apis/api/blob/main/src/structures/Connection.js#L341-L343)). Something like this:

```js
sendReconnect = () => {
	incrementStat('reconnects')
	this.send('RECONNECT')
}
```

That means we'll also have to add special handling for the reconnect message. We can do that in [`handlePRIVMessage`](), right after [line 45](https://github.com/fdgt-apis/api/blob/main/src/helpers/handlePRIVMSGMessage.js#L45) where we parse the message:

```js
if (command.toLowerCase().trim() === 'reconnect') {
	return connection.sendReconnect('RECONNECT')
}
```

**Comment by pajlada** [comment]
@trezy Thanks for your feedback! I implemented the code changes as you suggested and the tests still work! :slightly_smiling_face:

File: src/helpers/handlePRIVMSGMessage.js [hunk 1]
```diff
@@ -44,6 +44,10 @@ export default (messageData, connection) => {
 		...args
 	} = mri(argv)
 
+	if (command.toLowerCase().trim() === 'reconnect') {
+		return connection.sendReconnect('RECONNECT')
```
Comment (trezy) [inline src/helpers/handlePRIVMSGMessage.js:48]: One last nitpick: `sendReconnect` doesn't take any arguments, so we can remove the string that gets passed in.

**Review by trezy — CHANGES_REQUESTED** [review]
Added one last comment, but the rest of it looks great! Once you clean that up we can merge.

**Review by trezy — APPROVED** [review]
LGTM!

**Pull request merged**